src/renderer/components/app/navigation_bar/navigation_bar.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { COMPONENTS_LIST } from '../../../constants/componentsEnum';
+import { updateCurrentView } from '../../../../shared/slices/currentViewSlice';
+import './navigation_bar.scss';
+
+
+
+
+const getKeyValue = (key: string) => (obj: Record<string, any>) => obj[key];
+
+const NavigationBar = () => {
+
+  const dispatch = useDispatch();
+
+  const onNavigationButtonClick = (buttonKey: string) => () => {
+    dispatch(updateCurrentView(buttonKey))
+  }
+  
+  const renderListOfButtons = () => {
+    var listOfButtons:JSX.Element[] = [];
+    Object.keys(COMPONENTS_LIST).forEach((key) => {
+      const buttonKey = getKeyValue(key)(COMPONENTS_LIST);
+      listOfButtons.push(<button onClick={onNavigationButtonClick(buttonKey)}>{buttonKey}</button>)
+    })
+    return listOfButtons;
+  }
```

Comment:
Could just `map` over the keys in cases like these instead of `forEach` so that there is no need for any variables.

```tsx
const renderListOfButtons = () => {
    return Object.keys(COMPONENTS_LIST).map((key) => {
      const buttonKey = getKeyValue(key)(COMPONENTS_LIST);
      return (
        <button key={key} onClick={onNavigationButtonClick(buttonKey)}>
          {buttonKey}
        </button>
      );
    });
  };
```

One more thing is that if you iterate over an object or an array to return elements React requires those elements to have a unique `key` (error must have been shown in dev tools) so in this case it can just be the component name as it's unique but in cases where you don't have such key you can use index of the mapped element.